sim/core/cast.go

```diff
@@ -142,12 +142,9 @@ func (cast *Cast) startCasting(sim *Simulation, onCastComplete OnCastComplete) b
 	if cast.CastTime == 0 {
 		cast.internalOnComplete(sim, onCastComplete)
 	} else {
-		cast.Character.HardcastAura = Aura{
```

Comment:
I think the 'right' way to do this is to make a pending action here. Very similar to the dot logic except it only fires once.

For ele shaman, which doesnt use dots and therefore gets to use the shortcut for # pending actions = 1 in the main loop, adding a pending action for hardcasts might be a perf loss. But I'm not sure.

Reply:
I would guess its a perf loss, I can measure it though. Perhaps as a followup to this.